### PR TITLE
Allow option change tmp directory

### PR DESCRIPTION
**Pull request by faishal**
- Added env variable to change defualt `/tmp` location.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -5,10 +5,11 @@ const { onAdd, onCheck, onPush, onOpenPull, onUpdatePull } = require( './hooks'
...
-	console.log( 'Cleaning up /tmp' );
+	const TEMP_DIR = process.env.TEMP_DIR || '/tmp'
+	console.log( `Cleaning up ${ TEMP_DIR }` );
 	const rmrf = pify( rimraf );
-	await rmrf( '/tmp/downloads' );
-	await rmrf( '/tmp/repos' );
+	await rmrf( `${ TEMP_DIR }/downloads` );
+	await rmrf( `${ TEMP_DIR }/repos` );
 };
 const withClean = func => ( ...args ) => func( ...args ).then( () => clean() );
```
Comment (rmccue) [inline src/index.js:14]:
```suggestion
const withClean = func => ( ...args ) => func( ...args ).finally( () => clean() );
```

File: src/hooks.js [hunk 2]
```diff
@@ -120,7 +120,18 @@ const onCheck = async context => {
 	// Start a "build".
 	const { github, payload } = context;
 
-	const { head_branch, head_sha, pull_requests } = payload.check_suite || payload.check_run.check_suite;
+	let id, head_branch, head_sha, pull_requests;
+
+	if( payload.check_suite ) {
+		( { id, head_branch, head_sha, pull_requests } = payload.check_suite || payload.check_run.check_suite );
```
Comment (rmccue) [inline src/hooks.js:126]: Not sure why you've got bonus parens around these, is that required for the syntax?

File: src/hooks.js [hunk 3]
```diff
@@ -120,7 +120,18 @@ const onCheck = async context => {
...
+	} else if ( payload.check_run && payload.check_run.check_suite ){
+		( { id, head_branch, head_sha, pull_requests } = payload.check_run.check_suite );
+	} else if( payload.pull_request ) {
```
Comment (rmccue) [inline src/hooks.js:129]:
```suggestion
	} else if ( payload.pull_request ) {
```

File: src/hooks.js [hunk 4]
```diff
@@ -120,7 +120,18 @@ const onCheck = async context => {
...
+		pull_requests = [ payload.pull_request ];
+		head_branch = payload.pull_request.head.ref;
+		head_sha = payload.pull_request.head.sha;
+		id = payload.pull_request.id;
+	}
```
Comment (rmccue) [inline src/hooks.js:134]: What if none of these match? Need an `else` to fail here.

File: src/index.js [hunk 5]
```diff
@@ -1,21 +1,48 @@
...
-	// Clean up the tmp directories.
-	console.log( 'Cleaning up /tmp' );
+const TEMP_DIR = process.env.TEMP_DIR || '/tmp'
+const clean = async ( context ) => {
+	const { payload } = context;
+	let id, head_sha;
+
+	if ( payload.check_suite ) {
+		( { head_sha, id } = payload.check_suite );
+	} else if ( payload.check_run && payload.check_run.check_suite ){
```
Comment (rmccue) [inline src/index.js:16]:
```suggestion
	} else if ( payload.check_run && payload.check_run.check_suite ) {
```

File: src/index.js [hunk 6]
```diff
@@ -1,21 +1,48 @@
...
+		( { head_sha, id } = payload.check_run.check_suite );
+	} else if( payload.pull_request ){
```
Comment (rmccue) [inline src/index.js:18]:
```suggestion
	} else if ( payload.pull_request ) {
```

File: src/index.js [hunk 7]
```diff
@@ -1,21 +1,48 @@
...
+		head_sha = payload.pull_request.head.sha;
+		id = payload.pull_request.id;
+	}
```
Comment (rmccue) [inline src/index.js:21]: Needs an `else` here too.

**Review by rmccue — CHANGES_REQUESTED** [review]

File: src/hooks.js (hunk 2 above)
Comment (faishal) [reply src/hooks.js:126]: Yes - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#syntax

Considering I have declared variable early

**Review by kadamwhite — APPROVED** [review]
Changes have been addressed and this has been deployed and tested as part of an Altis-side bot, so merging away

**Pull request merged**